### ISSUE TITLE
resource_keycloak_oidc_identity_provider does not require client authentication

### ISSUE THREAD

**Issue by adresan** [mrparkers/terraform-provider-keycloak#223]
During i start to write docs for the resource_keycloak_oidc_identity_provider, i play a little bit around with it and testing creation and import. I notice that its possible to create a oidc provider without setting the `Client Authentication` field. This field is required if you create a oidc provier over the UI. 

![image](https://user-images.githubusercontent.com/59478315/74481322-cedda980-4eb2-11ea-9804-401b4ae1409a.png)

In the code of the resource_keycloak_oidc_identity_provider.go, this map is named `extra_config` and its marked as optional:

```
"extra_config": {
	Type:     schema.TypeMap,
	Optional: true,
	},
}
```

It you import a oidc provider, you could see in the Terraform statefile that the Client Authentication is  a attribute inside this map named  `clientAuthMethod`:

```
"extra_config": {
  "acceptsPromptNoneForwardFromClient": "",
  "clientAuthMethod": "private_key_jwt"
},
```
The mapping of this is:

`client_secret_post`  Client secret send as basic auth
`client_secret_basic`  Client secret sent as basic auth
`client_secret_jwt `Client secret as jwt
`private_key_jwt` JTW signed with private key

In my opinion, clientAuthMethod should be required and accept only the given options.

**Comment by tomrutsaert** [comment]
Keep in mind this is a new field since keycloak 8, people are using this provider for older versiobs as well, and in older versions this field is not required. (In older versions keycloak only supports implicitly secret post) if i am not mistaken keycloak 8 will fallback to secret post if it is not set.

Furthermore other idp inmplementatuons this field might not be required. Imho there should be any required field in the generic idp resource. 
For example for the google idp token url you can not fill in in the ui, it is not used, still it is a required field in this generic resource provider. Same for other fields in custom idp providers i have.

(Typed on my smartphone, i will edit and correct any mistakes in this comment later today)

**Comment by adresan** [comment]
Thanks for the input. If we need this field marked as optional for compatible reasons with older version, i will give this instead a extra mark in the documentation. I try to make the documentation supportive as possible for the users so they dont have to figure out how extra_config is filled and what is the mapping since keycloak 8. 

I close this issue together with a doc PR during this weekend.

**Comment by adresan** [comment]
closed with PR #224
